### PR TITLE
Add string.convert.convert_urls APIs to pylibcudf

### PR DESCRIPTION
**Pull request by mroeschke**
## Description
Contributes to https://github.com/rapidsai/cudf/issues/15162

Also I believe the cpp docstrings were incorrect, but could use a second look.

## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

File: python/pylibcudf/pylibcudf/libcudf/strings/convert/convert_urls.pxd [hunk 1]
```diff
@@ -8,7 +8,7 @@ from pylibcudf.libcudf.column.column_view cimport column_view
 cdef extern from "cudf/strings/convert/convert_urls.hpp" namespace \
         "cudf::strings" nogil:
     cdef unique_ptr[column] url_encode(
-        column_view input_col) except +
+        column_view input) except +
```
Comment (brandon-b-miller) [inline python/pylibcudf/pylibcudf/libcudf/strings/convert/convert_urls.pxd:11]: @vyasr should we be marking these APIs all with `libcudf_exception_handler` as we go even if currently the API has no documented exception cases?

**Review by brandon-b-miller — APPROVED** [review]
One question otherwise LGTM

File: python/pylibcudf/pylibcudf/libcudf/strings/convert/convert_urls.pxd (hunk 1 above)
Comment (vyasr) [reply python/pylibcudf/pylibcudf/libcudf/strings/convert/convert_urls.pxd:11]: I think that's probably a good best practice yeah.

**Review by ttnghia — APPROVED** [review]

File: python/pylibcudf/pylibcudf/libcudf/strings/convert/convert_urls.pxd (hunk 1 above)
Comment (mroeschke) [reply python/pylibcudf/pylibcudf/libcudf/strings/convert/convert_urls.pxd:11]: xref https://github.com/rapidsai/cudf/issues/17036 to follow up in the future

**Pull request merged**